### PR TITLE
#959 HGG update hotspots

### PR DESCRIPTION
**Pull request by kgaonkar6**
⚠️  merge #1073  before this PR since we need an updated tp53 annotation

#### What scientific question is your analysis addressing?
Rerun HGG subtype with hotspot+consensus maf

#### What was your approach?
Just add hotspot maf `pbta-snv-scavenged-hotspots.maf.tsv.gz` from `hotspot-detection` to the snv calls used for subtyping.

#### What GitHub issue does your pull request address?
#959 

#### Which areas should receive a particularly close look?
NA

#### Is there anything that you want to discuss further?
NA

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
yes

#### What types of results are included (e.g., table, figure)?
tables

#### What is your summary of the results?
 - 4 samples got H3F3A mutations added from hotspot mafs:
PNOC: BS_6DT506HY ,  BS_AEEKJ1QK ,  BS_TYFD8ZMG
CBTN: BS_YB99Q6TN

  | Chromosome | Start_Position | End_Position | Strand | Variant_Classification | IMPACT | Tumor_Sample_Barcode | Hugo_Symbol | HGVSp_Short | SIFT | DOMAINS | caller
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1 | chr1 | 226064434 | 226064434 | + | Missense_Mutation | MODERATE | BS_TYFD8ZMG | H3F3A | p.K28M | deleterious_low_confidence(0.01) | Gene3D:1.10.20.10,PDB-ENSP_mappings:3av2.A,PDB-ENSP_mappings:3av2.E,PDB-ENSP_mappings:3muk.D,PDB-ENSP_mappings:3wtp.E,PDB-ENSP_mappings:4h9n.A,PDB-ENSP_mappings:4h9o.A,PDB-ENSP_mappings:4h9p.A,PDB-ENSP_mappings:4h9q.A,PDB-ENSP_mappings:4h9r.A,PDB-ENSP_mappings:4h9s.A,PDB-ENSP_mappings:4h9s.B,PDB-ENSP_mappings:4hga.B,PDB-ENSP_mappings:4n4i.B,PDB-ENSP_mappings:5ay8.A,PDB-ENSP_mappings:5ay8.E,PDB-ENSP_mappings:5b32.A,PDB-ENSP_mappings:5b32.E,PDB-ENSP_mappings:5b33.A,PDB-ENSP_mappings:5b33.E,PDB-ENSP_mappings:5dwq.F,PDB-ENSP_mappings:5dwq.G,PDB-ENSP_mappings:5dx0.F,PDB-ENSP_mappings:5dx0.G,PDB-ENSP_mappings:5dx0.H,PDB-ENSP_mappings:5dx0.I,PDB-ENSP_mappings:5kdm.A,PDB-ENSP_mappings:5x7x.A,PDB-ENSP_mappings:5x7x.E,Pfam_domain:PF00125,Prints_domain:PR00622,hmmpanther:PTHR11426,hmmpanther:PTHR11426:SF163,Superfamily_domains:SSF47113,mobidb-lite | mutect2;strelka2
2 | chr1 | 226064434 | 226064434 | + | Missense_Mutation | MODERATE | BS_AEEKJ1QK | H3F3A | p.K28M | deleterious_low_confidence(0.01) | Gene3D:1.10.20.10,PDB-ENSP_mappings:3av2.A,PDB-ENSP_mappings:3av2.E,PDB-ENSP_mappings:3muk.D,PDB-ENSP_mappings:3wtp.E,PDB-ENSP_mappings:4h9n.A,PDB-ENSP_mappings:4h9o.A,PDB-ENSP_mappings:4h9p.A,PDB-ENSP_mappings:4h9q.A,PDB-ENSP_mappings:4h9r.A,PDB-ENSP_mappings:4h9s.A,PDB-ENSP_mappings:4h9s.B,PDB-ENSP_mappings:4hga.B,PDB-ENSP_mappings:4n4i.B,PDB-ENSP_mappings:5ay8.A,PDB-ENSP_mappings:5ay8.E,PDB-ENSP_mappings:5b32.A,PDB-ENSP_mappings:5b32.E,PDB-ENSP_mappings:5b33.A,PDB-ENSP_mappings:5b33.E,PDB-ENSP_mappings:5dwq.F,PDB-ENSP_mappings:5dwq.G,PDB-ENSP_mappings:5dx0.F,PDB-ENSP_mappings:5dx0.G,PDB-ENSP_mappings:5dx0.H,PDB-ENSP_mappings:5dx0.I,PDB-ENSP_mappings:5kdm.A,PDB-ENSP_mappings:5x7x.A,PDB-ENSP_mappings:5x7x.E,Pfam_domain:PF00125,Prints_domain:PR00622,hmmpanther:PTHR11426,hmmpanther:PTHR11426:SF163,Superfamily_domains:SSF47113,mobidb-lite | mutect2;strelka2
3 | chr1 | 226064434 | 226064434 | + | Missense_Mutation | MODERATE | BS_6DT506HY | H3F3A | p.K28M | deleterious_low_confidence(0.01) | Gene3D:1.10.20.10,PDB-ENSP_mappings:3av2.A,PDB-ENSP_mappings:3av2.E,PDB-ENSP_mappings:3muk.D,PDB-ENSP_mappings:3wtp.E,PDB-ENSP_mappings:4h9n.A,PDB-ENSP_mappings:4h9o.A,PDB-ENSP_mappings:4h9p.A,PDB-ENSP_mappings:4h9q.A,PDB-ENSP_mappings:4h9r.A,PDB-ENSP_mappings:4h9s.A,PDB-ENSP_mappings:4h9s.B,PDB-ENSP_mappings:4hga.B,PDB-ENSP_mappings:4n4i.B,PDB-ENSP_mappings:5ay8.A,PDB-ENSP_mappings:5ay8.E,PDB-ENSP_mappings:5b32.A,PDB-ENSP_mappings:5b32.E,PDB-ENSP_mappings:5b33.A,PDB-ENSP_mappings:5b33.E,PDB-ENSP_mappings:5dwq.F,PDB-ENSP_mappings:5dwq.G,PDB-ENSP_mappings:5dx0.F,PDB-ENSP_mappings:5dx0.G,PDB-ENSP_mappings:5dx0.H,PDB-ENSP_mappings:5dx0.I,PDB-ENSP_mappings:5kdm.A,PDB-ENSP_mappings:5x7x.A,PDB-ENSP_mappings:5x7x.E,Pfam_domain:PF00125,Prints_domain:PR00622,hmmpanther:PTHR11426,hmmpanther:PTHR11426:SF163,Superfamily_domains:SSF47113,mobidb-lite | mutect2;strelka2;vardict
4 | chr1 | 226064434 | 226064434 | + | Missense_Mutation | MODERATE | BS_YB99Q6TN | H3F3A | p.K28M | deleterious_low_confidence(0.01) | Gene3D:1.10.20.10,PDB-ENSP_mappings:3av2.A,PDB-ENSP_mappings:3av2.E,PDB-ENSP_mappings:3muk.D,PDB-ENSP_mappings:3wtp.E,PDB-ENSP_mappings:4h9n.A,PDB-ENSP_mappings:4h9o.A,PDB-ENSP_mappings:4h9p.A,PDB-ENSP_mappings:4h9q.A,PDB-ENSP_mappings:4h9r.A,PDB-ENSP_mappings:4h9s.A,PDB-ENSP_mappings:4h9s.B,PDB-ENSP_mappings:4hga.B,PDB-ENSP_mappings:4n4i.B,PDB-ENSP_mappings:5ay8.A,PDB-ENSP_mappings:5ay8.E,PDB-ENSP_mappings:5b32.A,PDB-ENSP_mappings:5b32.E,PDB-ENSP_mappings:5b33.A,PDB-ENSP_mappings:5b33.E,PDB-ENSP_mappings:5dwq.F,PDB-ENSP_mappings:5dwq.G,PDB-ENSP_mappings:5dx0.F,PDB-ENSP_mappings:5dx0.G,PDB-ENSP_mappings:5dx0.H,PDB-ENSP_mappings:5dx0.I,PDB-ENSP_mappings:5kdm.A,PDB-ENSP_mappings:5x7x.A,PDB-ENSP_mappings:5x7x.E,Pfam_domain:PF00125,Prints_domain:PR00622,hmmpanther:PTHR11426,hmmpanther:PTHR11426:SF163,Superfamily_domains:SSF47113,mobidb-lite | lancet;mutect2;vardict

 - Updated Tert hotspot promoter mutation:

  | Chromosome | Start_Position | End_Position | Strand | Variant_Classification | IMPACT | Tumor_Sample_Barcode | Hugo_Symbol | HGVSp_Short | SIFT | DOMAINS | caller
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1 | chr5 | 1295113 | 1295113 | + | 5'Flank | MODIFIER | BS_KAD49R68 | TERT | . | . | . | mutect2;strelka2;vardict
2 | chr5 | 1295113 | 1295113 | + | 5'Flank | MODIFIER | BS_MSB1RDAJ | TERT | . | . | . | lancet;mutect2;vardict

 - Updated TP53 annotation from hotspot mutations :

  | Chromosome | Start_Position | End_Position | Strand | Variant_Classification | IMPACT | Tumor_Sample_Barcode | Hugo_Symbol | HGVSp_Short | SIFT | DOMAINS | caller
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1 | chr17 | 7673803 | 7673803 | + | Missense_Mutation | MODERATE | BS_02YBZSBY | TP53 | p.R273C | deleterious(0) | cd08367,hmmpanther:PTHR11447:SF6,hmmpanther:PTHR11447,Gene3D:2.60.40.720,Pfam_domain:PF00870,Superfamily_domains:SSF49417,Prints_domain:PR00386 | mutect2;strelka2;vardict
2 | chr17 | 7675124 | 7675124 | + | Missense_Mutation | MODERATE | BS_68TZMZH1 | TP53 | p.Y163C | deleterious(0) | cd08367,hmmpanther:PTHR11447:SF6,hmmpanther:PTHR11447,Gene3D:2.60.40.720,Pfam_domain:PF00870,Superfamily_domains:SSF49417,Prints_domain:PR00386 | lancet;mutect2;vardict
3 | chr17 | 7673826 | 7673826 | + | Missense_Mutation | MODERATE | BS_FK3B5SDH | TP53 | p.L265P | deleterious(0) | cd08367,hmmpanther:PTHR11447:SF6,hmmpanther:PTHR11447,Gene3D:2.60.40.720,Pfam_domain:PF00870,Superfamily_domains:SSF49417,Prints_domain:PR00386 | lancet;mutect2

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd:85]:
```suggestion
 - We will use the `pbta-snv-consensus-mutation.maf.tsv.gz` from `snv-callers` module which gathers calls that are present in all 3 callers (strelka2, mutect2, and lancet)
```

File: analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd:115]: Could you make the columns you're selecting a vector that you pass to `fread()` both times? Just thinking about if you were to want to add or drop a column you'd have to change it in two places.

File: analyses/molecular-subtyping-HGG/02-HGG-molecular-subtyping-subset-files.R
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-HGG/02-HGG-molecular-subtyping-subset-files.R:120]: Same comment about saving selected columns as a vector

**Review by jaclyn-taroni — COMMENTED** [review]
Could have probably waited to review this, but I noticed a couple things in the relevant files. I don't need to review, but would like @jharenza's approval before merging.

File: analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd
Comment (kgaonkar6) [reply analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd:115]: Ah yes, I'll add that! I did that in LGAT module  but not here , funny how my mind works :D

**Review by jharenza — CHANGES_REQUESTED** [review]
Hi @kgaonkar6. I went through this again since the _TERT_ promoter mutations came back up and had to go back to this [issue](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/249) to refresh my memory. It looks like we should only be capturing the promoter mutations here. I know that wasn't part of the initial scope, but I think it should be an easy update (only retain 5' Flank hotspot mutations, C228T or C250T as per [this paper](https://link.springer.com/article/10.1007%2Fs00401-013-1195-5)). I realize this was not clear in our initial ticket, so sorry about that.

Otherwise, I think this is in good shape.

**Comment by kgaonkar6** [comment]
I've updated the snv filters for TERT here https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1077/commits/2866317a37e21cd723c05348871fafa6fd31b77c. 

Since C228T or C250T was from hg19 coordinates I've used the hg38 locations as discussed here: https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/819#issuecomment-788243124

**Comment by jharenza** [comment]
> I've updated the snv filters for TERT here https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1077/commits/2866317a37e21cd723c05348871fafa6fd31b77c. 
> 
> 
> 
> Since C228T or C250T was from hg19 coordinates I've used the hg38 locations as discussed here: https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/819#issuecomment-788243124

Thanks @kgaonkar6! I'll take a look. We may also want to note whether TERT promoter mutations are exclusive to H3 WT samples or not, as the publication linked in the issue claimed. I couldn't assess before, but now it should be easy.

**Review by jharenza — APPROVED** [review]
Thanks, @kgaonkar6. I took a look and the _TERT_ update looks good. Out of curiosity, I wanted to know if any samples which were H3 mutant contain _TERT_ promoter mutations and interestingly, three do:

```
# A tibble: 3 x 3
  Kids_First_Participant_ID Kids_First_Biospecimen_ID pathology_diagnosis                             
  <chr>                     <chr>                     <chr>                                           
1 PT_5BWZA0NT               BS_BFDEZK1C               Supratentorial or Spinal Cord PNET              
2 PT_63F63QY0               BS_KAD49R68               High-grade glioma/astrocytoma (WHO grade III/IV)
3 PT_40HTPY49               BS_T7WMJ08W               High-grade glioma/astrocytoma (WHO grade III/IV)
```

I wanted to make a note of those here, as I am not sure this has been described in the literature before. I double checked and these _TERT_ mutations were called in either 4/4 or 3/4 (mutect2;strelka2;vardict) of the callers and range from VAF 0.11-0.48. 

This is good now to go from my end.

**Pull request merged**
